### PR TITLE
RIIR update lints: Generate deprecated lints

### PR DESCRIPTION
**Pull request by phansch**
The update script now also generates the 'register_removed' section in
`clippy_lints/src/lib.rs`.

Also, instead of using `let mut store ...`, I added a new identifier
line so that the replacement will continue to work in case `let mut
store ...` ever changes.

cc #2882

### CODE REVIEW

File: clippy_dev/src/lib.rs [hunk 1]
```diff
@@ -84,6 +85,23 @@ pub fn gen_changelog_lint_list(lints: Vec<Lint>) -> Vec<String> {
...
+/// Generates the 'register_removed' code in `./clippy_lints/src/lib.rs`.
+pub fn gen_deprecated(lints: Vec<Lint>) -> Vec<String> {
+    lints.iter()
+        .filter(|l| l.deprecation.is_some())
+        .map(|l| {
+            format!(
+                r#"    store.register_removed(
+        "{}",
+        "{}",
+    );"#,
```
Comment (phansch) [inline clippy_dev/src/lib.rs]: This multi-line literal feels... brittle?

I found [indoc](https://github.com/dtolnay/indoc) by @dtolnay which could make this a little bit nicer. However, the code is unit tested and I'm not sure if it's worth adding a new dependency.

Comment (phansch) [reply clippy_dev/src/lib.rs]: oh, yeah. I forgot to run the dogfood tests locally :(

Comment (matthiaskrgr) [reply clippy_dev/src/lib.rs]: It's not gonna catch that because pedantic lints are not (yet..) enabled I think.

Comment (flip1995) [reply clippy_dev/src/lib.rs]: > pedantic lints are not (yet..) enabled I think.

Yep https://github.com/rust-lang-nursery/rust-clippy/issues/3172#issuecomment-432608735

**Comment by flip1995** [comment]
LGTM!

bors r+

**Comment by flip1995** [comment]
Oh forgot that macos nightly is missing. r=me when available again

**Pull request merged**